### DISCUSSION TITLE
关于serverless服务的一些疑问

### DISCUSSION THREAD

**Discussion by corvo007** [Mrs4s/go-cqhttp#994] [Q&A] [answered]
刚刚看到最新版本支持了腾讯云的serverless服务，但是我有几个问题不是很清楚：
1.在腾讯云上添加的时候，配置项应该怎么填写（如下图）：
![QYWFFO${`P6P5OYM~H5KPMP](https://user-images.githubusercontent.com/67192165/128471098-2e088cf7-b1eb-4810-b2fe-0ef74cca7181.png)
2.初次登录的验证应该怎么进行？还是说先在本地进行验证，然后将设备文件复制到serverless存储空间内？
3.如果程序因为封号等原因退出了，serverless服务会自动拉起吗？
谢谢！

**Comment by wdvxdr1123** [comment, marked as answer]
https://github.com/Mrs4s/go-cqhttp/blob/master/docs/config.md#%E4%BA%91%E5%87%BD%E6%95%B0%E9%83%A8%E7%BD%B2

**Comment by monsterxcn** [comment]
楼主现在有没有部署成功呢？还是这些配置项的问题，感觉文档写的太简洁了 2333

我今天尝试了一下 `go-cqhttp_1.0.0-beta5_linux_amd64` 似乎没有成功，测试使用 Api Gateway 事件模板稍作修改后失败，返回如下：

```
测试结果：
{
	"errorCode": 1,
	"errorMessage": "",
	"requestId": "64fa2ea0-b9c2-4551-bc17-54cd09ff670d",
	"statusCode": 443
}

SCF 日志：
START RequestId:64fa2ea0-b9c2-4551-bc17-54cd09ff670d
ERROR RequestId:64fa2ea0-b9c2-4551-bc17-54cd09ff670d Result:{"errorCode":-1,"errorMessage":"UserCodeError","statusCode":443}
END RequestId:64fa2ea0-b9c2-4551-bc17-54cd09ff670d
Report RequestId:64fa2ea0-b9c2-4551-bc17-54cd09ff670d Duration:0ms Memory:2048MB MemUsage:0.000000MB
```

以下是我的部署过程：

1. 在 CentOS7 上使用 `go-cqhttp_linux_amd64.tar.gz` 解压后首次执行 `./go-cqhttp`
2. 根据提示仅选择了「5: 云函数服务」生成 `config.yml`，然后填入账号信息在服务器上登录成功
   此时服务器上可以看到 `lambda 初始化失败: Post "http:///runtime/init/ready": http: no Host in request URL`
3. 将 `config.yml` `device.json` `bootstrap` 和 `go-cqhttp` 一起打包成 zip，内部仅此文件无二级文件夹
   不知道此处文件权限是否影响，是在 root 用户下操作的
4. SCF 自定义创建事件函数「基础配置」与楼主一致，未添加高级设置（环境变量在这里也没添加、权限配置默认未启用运行角色），代码由本地 zip 上传，执行方法尝试过 `go-cqhttp` `bootstrap`，触发器配置选用API网关触发方式且启用了集成响应
5. 测试事件是根据默认 Api Gateway 事件模板修改的，`body` 为空，有无 `access_token`、path `/runtime/init/ready` 的情况都试过：
   ![Api Gateway 事件](https://cdn.jsdelivr.net/gh/monsterxcn/MyBucket/issue/QQ截图20210814000924.png)


（在 SCF 编辑器终端试了下手动运行 go-cqhttp，看到是 lambda 初始化失败，经群友指点得知这个终端是没有云函数环境的。

**Reply by Huiaei** [reply]
解决了吗？

**Reply by monsterxcn** [reply]
没有，没搞明白

**Comment by Huiaei** [comment]
楼主部署成功生效了吗？
